firestore/query.go

```diff
@@ -250,6 +280,147 @@ func (q *Query) processCursorArg(name string, docSnapshotOrFieldValues []interfa
 
 func (q Query) query() *Query { return &q }
 
+func getSafeCursorValue(vProto *pb.Value, q Query) (interface{}, error) {
+	switch vProto.ValueType.(type) {
+	case *pb.Value_ReferenceValue:
+		refVal := vProto.GetReferenceValue()
+		if strings.HasPrefix(refVal, q.path) {
+			refVal = refVal[len(q.path)+1:]
+		}
+		return refVal, nil
+	default:
+		i, err := createFromProtoValue(vProto, q.c)
```

Comment:
`createFromProtoValue` should handle references as well. Did you try that?

Reply:
I don't think it does, at least as is. There is a todo item to support docRefs. I think this would clean things up though so I am taking the time to add that support.

Reply:
added allowing of documentRefs, so this code is greatly simplified.